pkg/subctl/cmd/show_connections.go

```diff
@@ -36,30 +36,35 @@ func getConnectionsStatus(config *rest.Config) []connectionStatus {
 	submarinerClient, err := submarinerclientset.NewForConfig(config)
 	exitOnError("Unable to get the Submariner client", err)
 
+	SubmarinerInstalled = true
```

Comment:
This seems like a weird way to implement conditional flow control, I'm sure you can do it without the variable.
In fact you can just return the empty list `status` and the code will work just fine.

Reply:
yes, it will work but the output when submariner is not installed won't be pretty.

```
$ ./../submariner-operator/bin/subctl show endpoints

Showing information for cluster "cluster3":
Submariner is not installed
subctl version: v0.8.0-18-g53608ad-dev
No resources found.

Showing information for cluster "cluster2":
CLUSTER ID                    ENDPOINT IP     PUBLIC IP       CABLE DRIVER        TYPE            
cluster2                      172.17.0.9      49.206.37.124   libreswan           local           

Showing information for cluster "cluster1":
Submariner is not installed
subctl version: v0.8.0-18-g53608ad-dev
No resources found.
```
Notice subctl version line before 'No resources found'

Whereas in current implementation, the output is
```
$ ./../submariner-operator/bin/subctl show endpoints

Showing information for cluster "cluster3":
Submariner is not installed
subctl version: v0.8.0-18-g53608ad-dev

Showing information for cluster "cluster2":
CLUSTER ID                    ENDPOINT IP     PUBLIC IP       CABLE DRIVER        TYPE            
cluster2                      172.17.0.9      49.206.37.124   libreswan           local           

Showing information for cluster "cluster1":
Submariner is not installed
subctl version: v0.8.0-18-g53608ad-dev
```

I guess it comes down to the way we want the output.

Reply:
I personally would prefer ```subctl version``` line to be the last which sort of says thats the end of information for the given cluster.

Reply:
There should be no `subctl version` printout at all as it's only printed out on error (and this is expected)

Reply:
I think it would be cleaner to verify if submariner is install at a higher level and not even call `getConnectionsStatus` and `printConnections`. Similar for endpoints et al. That would eliminate the need for `SubmarinerInstalled` and `show all` can bail early instead of printing "_Submariner is not installed_" multiple times. I suggest to create a function to retrieve the `*v1alpha1.Submariner` and pass it into `getConnectionsStatus`, `getEndpointsStatus` et al. 

In _show.go_:

```
func GetSubmarinerResource(config *rest.Config) *v1alpha1.Submariner {
	submarinerClient, err := submarinerclientset.NewForConfig(config)
	exitOnError("Unable to get the Submariner client", err)

	submariner, err := submarinerClient.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName, v1.GetOptions{})
	if err != nil {
		if apierrors.IsNotFound(err) {
			fmt.Println("Submariner is not installed")
			return nil
		}

		exitOnError("Error obtaining the Submariner resource", err)
	}

	return submariner
}

```
In _show_connections.go_:

```
func getConnectionsStatus(submariner *v1alpha1.Submariner) []connectionStatus {
	gateways := submariner.Status.Gateways
	...
}

func showConnections(cmd *cobra.Command, args []string) {
	...
	for _, item := range configs {
		fmt.Println()
		fmt.Printf("Showing information for cluster %q:\n", item.clusterName)

		showConnectionsFor(GetSubmarinerResource(item.config))
	}
}

func showConnectionsFor(submariner *v1alpha1.Submariner) {
	if submariner != nil {
		status := getConnectionsStatus(submariner)
		printConnections(status)
	}
}

```
In _show_all.go_:

```
func showAll(cmd *cobra.Command, args []string) {
	...
	for _, item := range configs {
		fmt.Println()
		fmt.Printf("Showing information for cluster %q:\n", item.clusterName)

                submariner := GetSubmarinerResource(item.config)
                if submariner == nil {
                    continue
                }

                 ...
		fmt.Println("\nShowing Connections details")
		showConnectionsFor(submariner)
		...
	}
}
```